### PR TITLE
Chore/improve telemetry

### PR DESCRIPTION
**Pull request by vanbasten17**
## Description
* Added `BOTONIC_DISABLE_ANALYTICS` secret to the repo and in its corresponding workflow
* Improved analytics by retrieving `@botonic/cli`'s version and `@botonic` dependencies.
* Refactored json logic and renamed linter suggestions.

## Context
`BOTONIC_DISABLE_ANALYTICS` should be declared in order to not track installations run by Github Actions, because they can add noise to the analytics.

## Next steps
Once this PR is approved, I'm gonna apply these changes also in `botonic-examples` track methods. Unifying and improving error messages in related @ericmarcos task: https://linear.app/hubtype/issue/BOT-171/better-understand-1st-time-developer-flow

### CODE REVIEW

**Comment by manuelfidalgo** [comment]
LGTM! :ok_hand:

**Review by manuelfidalgo — APPROVED** [review]
:ok_hand:

File: packages/botonic-cli/src/utils.ts [hunk 1]
```diff
@@ -77,19 +106,19 @@ export function track(event: string, properties = {}) {
...
   }
 }
 
-export function sleep(ms: number) {
+export function sleep(ms: number): Promise<number> {
   return new Promise(resolve => setTimeout(resolve, ms))
 }
 
-async function sh(cmd) {
+async function sh(cmd: string) {
```
Comment (dpinol) [inline packages/botonic-cli/src/utils.ts:121]: type return to 
`{ stdout: string, stderr: string }`

**Review by dpinol — CHANGES_REQUESTED** [review]

File: .github/workflows/botonic-cli-tests.yml [hunk 2]
```diff
@@ -30,7 +30,7 @@ jobs:
             ${{ runner.os }}-node-
       - name: Install dev dependencies
         env:
-          BOTONIC_DISABLE_MIXPANEL: 1
+          BOTONIC_DISABLE_ANALYTICS: ${{ secrets.BOTONIC_DISABLE_ANALYTICS }}
```
[the remark this answers is not in the document]
Comment (ericmarcos) [reply .github/workflows/botonic-cli-tests.yml:33]: agree

**Review by ericmarcos — APPROVED** [review]

File: .github/workflows/botonic-cli-tests.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (vanbasten17) [reply .github/workflows/botonic-cli-tests.yml]: sure, changed 👍

File: packages/botonic-cli/src/utils.ts [hunk 3]
```diff
@@ -1,35 +1,42 @@
...
-  if (!fs.existsSync(botonic_credentials_path)) {
+  if (!fs.existsSync(botonicCredentialsPath)) {
     initializeCredentials()
   }
   try {
-    credentials = JSON.parse(fs.readFileSync(botonic_credentials_path, 'utf8'))
+    credentials = readJSON(botonicCredentialsPath)
   } catch (e) {
-    if (fs.existsSync(botonic_credentials_path)) {
+    if (fs.existsSync(botonicCredentialsPath)) {
```
[the remark this answers is not in the document]
Comment (vanbasten17) [reply packages/botonic-cli/src/utils.ts:39]: good catch, this is unnecessary 👍

File: packages/botonic-cli/src/utils.ts [hunk 4]
```diff
@@ -1,35 +1,42 @@
...
       console.warn('Credentials could not be loaded', e)
```
[the remark this answers is not in the document]
Comment (vanbasten17) [reply packages/botonic-cli/src/utils.ts:40]: it is provided 👍

File: packages/botonic-cli/src/utils.ts [hunk 5]
```diff
@@ -41,26 +48,48 @@ try {
...
-if (analytics_enabled()) {
+if (isAnalyticsEnabled()) {
   analytics = new Analytics('YD0jpJHNGW12uhLNbgB4wbdTRQ4Cy1Zu')
 }
 
+function getBotonicCliVersion(): string | undefined {
+  try {
+    return String(execSync('botonic --version')).trim()
+  } catch (e) {
+    return undefined
```
[the remark this answers is not in the document]
Comment (vanbasten17) [reply packages/botonic-cli/src/utils.ts:63]: added 👍

**Pull request merged**